### ISSUE TITLE
model null 

### ISSUE THREAD

**Issue by pythoneer** [brianegan/scoped_model#15]
TLDR;
due to the wrong usage of imports, models can get "null"

i changed 
```dart
import 'model.dart';
```
to 
```dart
import 'package:scoped_model_test/model.dart';
```

and it works now. 

Original Post:

i think its related to this issue but i can't really figure out how to do it anyway. 

https://github.com/brianegan/scoped_model/issues/10

The simple one file example works fine, but as soon as i start to refactor things out across multiple files things start to crash. 

For simplicity i just took the initial example and split it to separate files

- main.dart
- MainSreen.dart
- model.dart

all under the lib folder.

main.dart:
```dart
import 'model.dart';
import 'package:flutter/material.dart';
import 'package:scoped_model/scoped_model.dart';
import 'package:scoped_model_test/MainScreen.dart';

void main() {
  runApp(new MyApp());
}

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => new _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return new ScopedModel<CounterModel>(
      model: CounterModel(),
      child: new MaterialApp(
        home: MainScreen(),
      ),
    );
  }
}

``` 

MainScreen.dart
```dart
import 'model.dart';

import 'package:flutter/material.dart';
import 'package:scoped_model/scoped_model.dart';

class MainScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Material(
      child: new Padding(
        padding: const EdgeInsets.all(28.0),
        child: Column(
          children: <Widget>[
//            ScopedModelDescendant<CounterModel>(builder:(context, child, model) { return Text(model.counter.toString());}),
            ScopedModelDescendant<CounterModel>(
                builder:(BuildContext context, Widget child, CounterModel model) {
                  print("model: $model");
                  return MaterialButton(
                    child: Text("+"),
                    onPressed: () {print("click"); /*model.increment();*/},
                  );
                }
            )
          ],
        ),
      ),
    );
  }
}

``` 

model.dart
```dart
import 'package:scoped_model/scoped_model.dart';

class CounterModel extends Model {
  int _counter = 0;
  int get counter => _counter;
  void increment() {
    _counter++;
    notifyListeners();
  }
}
``` 

i expected to just missed an import due to the description in the other issue, but i don't think that's the case here, but i think it is somehow related.

EDIT:
The code above is coming from a reduced example i did for myself. If i 100% apply this to the original code from the example code https://github.com/brianegan/scoped_model/blob/master/example/lib/main.dart  it works fine ... but i can't really spot the essential difference :/

working code

main.dart
```dart
import 'package:flutter/material.dart';
import 'package:scoped_model/scoped_model.dart';
import 'package:scoped_model_test/model.dart';
import 'package:scoped_model_test/CounterHome.dart';

void main() {
  runApp(new MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new ScopedModel<CounterModel>(
      model: new CounterModel(),
      child: new MaterialApp(
        title: 'Flutter Demo',
        theme: new ThemeData(
          primarySwatch: Colors.green,
        ),
        home: new CounterHome('Scoped Model Demo'),
      ),
    );
  }
}

```

model.dart
```dart
import 'package:scoped_model/scoped_model.dart';

class CounterModel extends Model {
  int _counter = 0;
  int get counter => _counter;
  void increment() {
    _counter++;
    notifyListeners();
  }
}
```

CounterHome.dart
```dart
import 'package:flutter/material.dart';
import 'package:scoped_model_test/model.dart';
import 'package:scoped_model/scoped_model.dart';

class CounterHome extends StatelessWidget {
  final String title;

  CounterHome(this.title);

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text(title),
      ),
      body: new Center(
        child: new Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            new Text(
              'You have pushed the button this many times:',
            ),
            new ScopedModelDescendant<CounterModel>(
              builder: (context, child, model) => new Text(
                  model.counter.toString(),
                  style: Theme.of(context).textTheme.display1),
            ),
          ],
        ),
      ),
      floatingActionButton: new ScopedModelDescendant<CounterModel>(
        builder: (context, child, model) => new FloatingActionButton(
          onPressed: model.increment,
          tooltip: 'Increment',
          child: new Icon(Icons.add),
        ),
      ),
    );
  }
}
```

EDIT2:

Ok i just figured out, that this had something to do with how the imports are made. I just followed the autocompletion in Android Studio and changed 

```dart
import 'model.dart';
```
to 
```dart
import 'package:scoped_model_test/model.dart';
```

i think i have to look up what this actually means in regards to dart (i am quite new to that language). I think i can close this for now and added a TLDR; on top.

**Comment by brianegan** [comment]
@pythoneer Hey, no worries, glad ya figured it out! Ya, for some reason, Dart treats those two files differently, but I haven't found a good explanation as to why it's doing that :( 

I just always use the full path to my files as you've done and everything seems to work, but this is literally the Top error I see across all my projects!

**Comment by smeetbhatt1** [comment]
How will I write the below code if I got more than two scoped model in the app?

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new ScopedModel<**CounterModel>(
      model: new CounterModel(),**
